pennylane/devices/qutrit_mixed/apply_operation.py

```diff
@@ -0,0 +1,201 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Functions to apply operations to a qutrit mixed state."""
+# pylint: disable=unused-argument
+
+from functools import singledispatch
+from string import ascii_letters as alphabet
+import pennylane as qml
+from pennylane import math
+from pennylane import numpy as np
+from pennylane.operation import Channel
+from .utils import QUDIT_DIM, get_einsum_mapping, get_new_state_einsum_indices
+
+alphabet_array = np.array(list(alphabet))
+
+
+def apply_operation_einsum(op: qml.operation.Operator, state, is_state_batched: bool = False):
+    r"""Apply a quantum channel specified by a list of Kraus operators to subsystems of the
+    quantum state. For a unitary gate, there is a single Kraus operator.
+
+    Args:
+        op (Operator): Operator to apply to the quantum state
+        state (array[complex]): Input quantum state
+        is_state_batched (bool): Boolean representing whether the state is batched or not
+
+    Returns:
+        array[complex]: output_state
+    """
+    einsum_indices = get_einsum_mapping(op, state, _map_indices, is_state_batched)
+
+    num_ch_wires = len(op.wires)
+    kraus = _get_kraus(op)
+
+    # Shape kraus operators
+    kraus_shape = [len(kraus)] + [QUDIT_DIM] * num_ch_wires * 2
+    # Compute K^T, will be list of lists if broadcasting
+    kraus_transpose = []
+    if not isinstance(op, Channel):
+        # TODO Channels broadcasting doesn't seem to be implemented for qubits, should they be for qutrit?
+        mat = op.matrix()
+        dim = QUDIT_DIM**num_ch_wires
+        batch_size = math.get_batch_size(mat, (dim, dim), dim**2)
+        if batch_size is not None:
+            # Add broadcasting dimension to shape
+            kraus_shape = [batch_size] + kraus_shape
+            if op.batch_size is None:
+                op._batch_size = batch_size  # pylint:disable=protected-access
+            for op_mats in kraus:
+                kraus_transpose.append([math.transpose(k) for k in op_mats])
+    if not kraus_transpose:
+        kraus_transpose = [math.transpose(k) for k in kraus]
+
+    kraus = math.stack(kraus)
+    kraus_transpose = math.stack(kraus_transpose)
+    # Torch throws error if math.conj is used before stack
+    kraus_dagger = math.conj(kraus_transpose)
+
+    kraus = math.cast(math.reshape(kraus, kraus_shape), complex)
+    kraus_dagger = math.cast(math.reshape(kraus_dagger, kraus_shape), complex)
+
+    return math.einsum(einsum_indices, kraus, state, kraus_dagger)
+
+
+def _map_indices(**kwargs):
```

Comment:
This function is different from the one that's put in the utils file it is specific to this einsum. It may be better to change the name to `_map_indices_apply_channel` or something similar. It was originally defined inside `apply_operation_einsum` but I was told to move it.